snuba/utils/streams/kafka.py

```diff
@@ -257,61 +283,41 @@ def __seek(self, offsets: Mapping[TopicPartition, int]) -> None:
                     ConfluentTopicPartition(stream.topic, stream.partition, offset)
                 )
 
-        self.__offsets.update(offsets)
+        self.__working_offsets.update(offsets)
 
     def seek(self, offsets: Mapping[TopicPartition, int]) -> None:
         if self.__state in {KafkaConsumerState.CLOSED, KafkaConsumerState.ERROR}:
             raise InvalidState(self.__state)
 
-        if offsets.keys() - self.__offsets.keys():
+        if offsets.keys() - self.__working_offsets.keys():
             raise ConsumerError("cannot seek on unassigned streams")
 
         self.__seek(offsets)
 
-    def commit(self) -> Mapping[TopicPartition, int]:
+    def stage_offsets(self, offsets: Mapping[TopicPartition, int]) -> None:
         if self.__state in {KafkaConsumerState.CLOSED, KafkaConsumerState.ERROR}:
             raise InvalidState(self.__state)
 
-        result: Optional[Sequence[ConfluentTopicPartition]] = None
+        if offsets.keys() - self.__working_offsets.keys():
+            raise ConsumerError("cannot stage offsets for unassigned streams")
 
-        retries_remaining = 3
-        while result is None:
-            try:
-                result = self.__consumer.commit(asynchronous=False)
-                assert result is not None
-            except KafkaException as e:
-                if not e.args[0].code() in (
-                    KafkaError.REQUEST_TIMED_OUT,
-                    KafkaError.NOT_COORDINATOR_FOR_GROUP,
-                    KafkaError._WAIT_COORD,
-                ):
-                    raise
-
-                if not retries_remaining:
-                    raise
-
-                logger.warning(
-                    "Commit failed: %s (%d retries remaining)",
-                    str(e),
-                    retries_remaining,
-                )
-                retries_remaining -= 1
-                time.sleep(1)
```

Comment:
I don't think this was the appropriate level of abstraction for this. Opinions?

Reply:
To be fair I am not sure why we retry in the kafka parent class transparently to the callsite without allowing the caller to control this behavior.

Reply:
I'll move this to the batching consumer.